mautic/mautic
# More descriptive API errors

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Enhancement
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The API error messages aren't too user friendly. This PR describes better why the error occurred and how to resolve it.

#### Steps to test this PR:
1. Disable API in Mautic configuration. Try to access an API endpoint and you'll get

> Unauthorized access to requested URL: /api/segments?limit=50

Whit this PR you'll see

> API disabled. You need to enable the API in the API settings page of your Mautic account.

2. Enable API, but leave Basic Auth disabled. Try to access API with Basic Auth. Before the PR you get error message:

> API authorization denied.

Whit this PR you'll see

> Basic Auth disabled. You need to enable HTTP basic auth in the API settings page of your Mautic account.

3. Enable API and Basic Auth, but try to log in with invalid credentials. The old message was:

> API authorization denied.

The new one is:

> Authorization denied, invalid credentials.

4. Make sure you can get the correct response with correct credentials.
